drivers/at86rf2xx/include/at86rf2xx_registers.h

```diff
@@ -532,6 +532,7 @@ extern "C" {
 #define AT86RF2XX_TRX_RPC_MASK__PLL_RPC_EN                      (0x08)
 #define AT86RF2XX_TRX_RPC_MASK__XAH_TX_RPC_EN                   (0x04)
 #define AT86RF2XX_TRX_RPC_MASK__IPAN_RPC_EN                     (0x02)
+#define AT86RF2XX_TRX_RPC_MASK__RX_RPC__CTRL_DISABLED           (0xC1)
```

Comment:
The CI is uphappy that this is undocumented (the CI only enforces this on new code (via an ugly hack), that's why the CI is not frowning upon the other lines). Since you have to touch this anyway to get this past the CI, how about this instead:

```suggestion
/**
 * @brief   Bits to set to enable smart idle
 */
#define AT86RF2XX_TRX_RPC_MASK__RX_RPC__SMART_IDLE \
        (AT86RF2XX_TRX_RPC_MASK__RX_RPC_EN \
        | AT86RF2XX_TRX_RPC_MASK__PDT_RPC_EN \
        | AT86RF2XX_TRX_RPC_MASK__PLL_RPC_EN \
        | AT86RF2XX_TRX_RPC_MASK__XAH_TX_RPC_EN \
        | AT86RF2XX_TRX_RPC_MASK__IPAN_RPC_EN)
```

Than this could be used to enable smart idle:

```C
void at86rf2xx_enable_smart_idle(at86rf2xx_t *dev)
{
#if AT86RF2XX_SMART_IDLE_LISTENING
    uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_RPC);
    tmp |= AT86RF2XX_TRX_RPC_MASK__RX_RPC__SMART_IDLE;
    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_RPC, tmp);
    at86rf2xx_set_rxsensitivity(dev, RSSI_BASE_VAL);
#else
    (void) dev;
#endif
}
```

and to disable it:

```C
void at86rf2xx_disable_smart_idle(at86rf2xx_t *dev)
{
    #if AT86RF2XX_SMART_IDLE_LISTENING
    uint8_t tmp = at86rf2xx_reg_read(dev, AT86RF2XX_REG__TRX_RPC);
    tmp &= ~AT86RF2XX_TRX_RPC_MASK__RX_RPC__SMART_IDLE;
    at86rf2xx_reg_write(dev, AT86RF2XX_REG__TRX_RPC, tmp);
    #else
        (void) dev;
    #endif
}
```

Btw: Feel free to squash afterwards.